test/test_hpylong.py

```diff
@@ -1,8 +1,22 @@
 from .support import HPyTest
 
+
 class TestLong(HPyTest):
 
-    def test_long_aslong(self):
+    def test_Long_FromLong(self):
```

Comment:
I know that you are simply following the existing tests and might be a chance which is worth its own PR, but I think that at some point we should start to test these API more: e.g., what happens if you pass something of the wrong type, an `int` subclass, a class which implements `__int__` and/or `__index__`, etc.

Reply:
For `HPyLong_From...` one can't really pass the wrong thing (because it won't compile) but let me at least add a simple check to each of the `HPyLong_As...` so that we're sure it does the right thing for non-number types.